community/kernel/src/main/java/org/neo4j/tooling/GlobalGraphOperations.java

```diff
@@ -155,12 +160,37 @@ protected Relationship fetchNextOrNull()
      */
     public Iterable<RelationshipType> getAllRelationshipTypes()
     {
+        return getAllRelationshipTypes( false );
+    }
+
+    public Iterable<RelationshipType> getAllRelationshipTypesInUse()
+    {
+        return getAllRelationshipTypes( true );
+    }
+
+    private Iterable<RelationshipType> getAllRelationshipTypes( boolean inUse )
+    {
         assertInUnterminatedTransaction();
-        return cast( relationshipTypes.getAllTokens() );
+        Iterable<RelationshipTypeToken> relationshipTypes = this.relationshipTypes.getAllTokens();
+        if ( inUse )
+        {
+            relationshipTypes = filter( new Predicate<Token>()
+            {
+                @Override
+                public boolean test( Token token )
+                {
+                    Statement statement = statementCtxSupplier.get();
```

Comment:
Can we instead of counting number of relationships for each separate token collect all tokens count store is aware of and thread them as used?
Almost the same question for labels case.